docs/RtspStreamingGuide.md

```diff
@@ -0,0 +1,371 @@
+   
+   ```bash
+   curl -X POST -H 'Content-Type: application/json'  \
+   http://localhost:59881/api/v2/device \
+   -d '[
+            {
+               "apiVersion": "v2",
+               "device": {
+                  "name":"Camera001",
+                  "serviceName": "device-usb-camera",
+                  "profileName": "USB-Camera-General",
+                  "description": "My test camera",
+                  "adminState": "UNLOCKED",
+                  "operatingState": "UP",
+                  "protocols": {
+                	  "USB": {
+                    	"CardName": "NexiGo N930AF FHD Webcam: NexiG",
+                    	"Path": "/dev/video6",
+ 			                "AutoStreaming": "false"
+                     }
+                  }
+               }
+            }
+   ]'
+   ```
+
+   Example Output: 
+   ```bash
+   [{"apiVersion":"v2","statusCode":201,"id":"fb5fb7f2-768b-4298-a916-d4779523c6b5"}]
+   ```
+
+## Start Video Streaming
+Unless the device service is configured to stream video from the camera automatically, a 'StartStreaming' command must be sent to the device service.
+
+There are two types of options:
+- The options start with **Input** prefix are used for the camera, such as specifying the image size and pixel format.
+- The options start with **Output** prefix are used for the output video, such as specifying aspect ratio and quality.
+
+These options can be passed in through Object value when calling StartStreaming.
+
+Query parameter:
+- **DeviceName**: The name of the camera
+
+For example:
+```shell
+curl -X PUT -d '{
+    "StartStreaming": {
+      "InputImageSize": "640x480",
+      "OutputVideoQuality": "5"
+    }
+}' http://localhost:59882/api/v2/device/name/<device name>/StartStreaming
+```
+
+Supported Input options:
+- **InputFps**: Ignore original timestamps and instead generate timestamps assuming constant frame rate fps. (default - same as source)
+- **InputImageSize**: Specifies the image size of the camera. The format is `wxh`, for example "640x480". (default - automatically selected by FFmpeg)
+- **InputPixelFormat**: Set the preferred pixel format (for raw video). (default - automatically selected by FFmpeg)
+
+Supported Output options:
+- **OutputFrames**: Set the number of video frames to output. (default - no limitation on frames)
+- **OutputFps**: Duplicate or drop input frames to achieve constant output frame rate fps. (default - same as InputFps)
+- **OutputImageSize**: Performs image rescaling. The format is `wxh`, for example "640x480". (default - same as InputImageSize)
+- **OutputAspect**: Set the video display aspect ratio specified by aspect. For example "4:3", "16:9". (default - same as source)
+- **OutputVideoCodec**: Set the video codec. For example "mpeg4", "h264". (default - mpeg4)
+- **OutputVideoQuality**: Use fixed video quality level. Range is a integer number between 1 to 31, with 31 being the worst quality. (default - dynamically set by FFmpeg)
```

Comment:
To be consistent with onvif, I think that the bolds should be enclosed in `back quotes`

Reply:
do you mean like this: - `OutputFrames`:

Reply:
Yes